### ISSUE TITLE
The accuracy of 'fear' and 'disgust' are always zero in Meld datasets

### ISSUE THREAD

**Issue by mrFocusXin** [zerohd4869/MM-DFN#5]
The  accuracy of 'fear' and 'disgust' are always zero in Meld datasets, so I can't reproduce the result in paper.
The results:
![image](https://user-images.githubusercontent.com/41821856/231211926-9aa9539e-7bd6-44ee-98f1-8a9f99ee3244.png)

Is there something wrong with my understanding?

**Comment by jellyfishovo** [comment]
Greetings!
i dont know how to train on MELD dataset
i just change the parameter from IEMOCAP to MELD:
 parser.add_argument('--dataset', default='IEMOCAP', help='dataset to train and test')
but it still can't work
What should i do
Thanks

**Comment by mrFocusXin** [comment]
Thank you for your prompt reply!



I wish you all the best






------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年4月19日(星期三) 下午4:03
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [zerohd4869/MM-DFN] The accuracy of &#39;fear&#39; and &#39;disgust&#39; are always zero in Meld datasets (Issue #5)





  
The accuracy of 'fear' and 'disgust' are always zero in Meld datasets, so I can't reproduce the result in paper. The results: 
 
Is there something wrong with my understanding?
  
Hi,
 
Thank you for your interest. Regarding the issue you mentioned, the 'fear' and 'disgust' categories in the MELD dataset are easily misclassified and always get zero accuracy as they account for a small proportion in the training set. Our MMDFN doesn't specifically optimize for the imbalanced classes problem in MELD.
 
To improve the accuracy of 'fear' and 'disgust,' you can set the '--class_weight' flag to True when running the script. This applies class weights in the cross-entropy/focal loss to alleviate the imbalanced classes problem. However, this may lead to a slight decrease in overall performance. Additionally, another alternative is to use BERT/RoBERTa to process text features (instead of using TextCNN with Glove embedding as we did in the paper), and it can achieve better performance on each emotion class in most cases.
 
Furthermore, optimal hyper-parameters (i.e., the number of GCN layers) may differ under different experimental configurations, such as the version of CUDA and PyTorch. Therefore, during the reproduction process, you can try modifying the default hyper-parameter values provided in the script. To facilitate this, we have added suggested hyper-parameter search ranges in the latest *.sh script. You can use this to select the optimal hyper-parameters directly.
 
I hope this helps. Let me know if you have any further questions.
